libcontainer/nsenter/cloned_binary.c

```diff
@@ -298,6 +314,19 @@ enum {
 #  endif
 #endif
 
+static inline bool is_memfd_unsupported_error(int err)
+{
+	/* Handle -errno and errno. */
+	err = abs(err);
+	/*
+	 * ENOSYS is obviously an "unsupported" error, EINVAL could be hit if our
+	 * minimum flags are not supported or on older kernels with
+	 * vm.memfd_noexec=2, and EACCES is returned on newer kernels with
+	 * vm.memfd_noexec=2 (indicating executable memfds are disabled).
+	 */
```

Comment:
Can we determine the "older" and "newer" kernel version?

Reply:
The patch hasn't yet been merged into Linus' tree so I don't know which release it will end up in, but the (partial) fix for the broken vm.memfd_noexec=2 contains the EACCES change. I expect it will end up in 6.6 but we'll see.